### DISCUSSION TITLE
How to support custom types?

### DISCUSSION THREAD

**Discussion by razaqq** [fnc12/sqlite_orm#1254] [Q&A] [answered]
How can i add support for smth like this?

```c++
struct User
{
	std::unordered_set<int> Payments;
	std::chrono::year_month_day Birthday;
};

storage.insert(User{ ... });
```

**Comment by fnc12** [comment]
how would you encode `std::unordered_set<int>` into a single column with raw SQLite?

**Reply by razaqq** [reply]
lets say its a `vector` not a set, by stringifying it?

**Reply by fnc12** [reply, marked as answer]
ok I see. One can bind any custom type if you know how to map it to raw SQLite values and back. Check these examples
https://github.com/fnc12/sqlite_orm/blob/master/examples/enum_binding.cpp
https://github.com/fnc12/sqlite_orm/blob/master/examples/nullable_enum_binding.cpp

**Reply by razaqq** [reply]
alright perfect, thanks for the fast reply!
